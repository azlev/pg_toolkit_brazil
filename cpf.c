#include "postgres.h"
#include "fmgr.h"
#include "utils/palloc.h"

PG_MODULE_MAGIC;

PG_FUNCTION_INFO_V1(cpfout);

Datum
cpfout(PG_FUNCTION_ARGS)
{
	int64		val = PG_GETARG_INT64(0);
	int			group1 = (val / 100000000) % 1000;
	int			group2 = (val / 100000) % 1000;
	int			group3 = (val / 100) % 1000;
	int			group4 = (val % 100);
	char	   *result;

	result = psprintf("%03d.%03d.%03d-%02d", group1, group2, group3, group4);

	PG_RETURN_CSTRING(result);
}
