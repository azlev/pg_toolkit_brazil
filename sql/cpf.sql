CREATE EXTENSION pg_toolkit_brazil;

CREATE TABLE pessoa(cpf cpf);
\d pessoa
INSERT INTO pessoa VALUES (cpf '40100276300');
SELECT * FROM pessoa;
