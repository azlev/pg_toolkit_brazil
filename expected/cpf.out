CREATE EXTENSION pg_toolkit_brazil;
CREATE TABLE pessoa(cpf cpf);
\d pessoa
             Table "public.pessoa"
 Column | Type | Collation | Nullable | Default 
--------+------+-----------+----------+---------
 cpf    | cpf  |           |          | 

INSERT INTO pessoa VALUES (cpf '40100276300');
SELECT * FROM pessoa;
      cpf       
----------------
 401.002.763-00
(1 row)

